### PR TITLE
Allow different ssh key file names than the default id_rsa

### PR DESCRIPTION
**Pull request by iljarotar**
Also added new docker installation to this PR so all of the changes can be tested with an image override. Will be split afterwards.

### CODE REVIEW

File: partition/roles/mgmt-server/tasks/main.yaml [hunk 1]
```diff
@@ -39,11 +39,26 @@
     name:
     - net-tools
     - ipmitool
-    - docker.io
```
Comment (Gerrit91) [inline partition/roles/mgmt-server/tasks/main.yaml:42]: Unrelated changes, cannot approve. :sweat_smile:

Comment (iljarotar) [reply partition/roles/mgmt-server/tasks/main.yaml]: removed

**Review by Gerrit91 — APPROVED** [review]
